controllers/machine.go

```diff
@@ -326,6 +328,27 @@ func (mrc *machineReconcileContext) ensureHardwareUserData(hardware *tinkv1.Hard
 	return nil
 }
 
+func (mrc *machineReconcileContext) ensureHardwareDiskData(hardware *tinkv1.Hardware) error {
+       foundDisks := make([]tinkv1.Disk, 0)
+       metadata := hardware.Spec.Metadata
+       if metadata != nil {
+			instances := metadata.Instance
+			if instances != nil {
+					storage := metadata.Instance.Storage
+					if storage != nil {
+							disks := metadata.Instance.Storage.Disks
+							diskLen := len(disks)
+							for i := 0; i < diskLen; i++ {
+									foundDisks = append(foundDisks, tinkv1.Disk{Device: disks[i].Device})
+							}
```

Comment:
you can `range` over the disks here
```suggestion
							for _, disk := range metadata.Instance.Storage.Disks {
									foundDisks = append(foundDisks, tinkv1.Disk{Device: disk.Device})
							}
```